src/analysis/complex/liouville.lean

```diff
@@ -0,0 +1,106 @@
+are formalized in `complex.apply_eq_apply_of_differentiable_of_bounded`,
+`complex.exists_const_forall_eq_of_differentiable_of_bounded`, and
+`complex.exists_eq_const_of_differentiable_of_bounded`.
+
+The proof is based on the Cauchy integral formula for the derivative of an analytic function, see
+`complex.deriv_eq_smul_circle_integral`.
+-/
+
+open topological_space metric set filter asymptotics function measure_theory
+open_locale topological_space filter nnreal real
+
+universes u v
+variables {E : Type u} [normed_group E] [normed_space ℂ E]
+  {F : Type v} [normed_group F] [normed_space ℂ F] [measurable_space F] [borel_space F]
+    [second_countable_topology F] [complete_space F]
+
+namespace complex
+
+/-- If `f` is complex differentiable on a closed disc with center `c` and radius `R > 0`, then
+`f' c` can be represented as an integral over the corresponding circle.
+
+TODO: add a version for `w ∈ metric.ball c R`.
+
+TODO: add a version for higher derivatives. -/
+lemma deriv_eq_smul_circle_integral {R : ℝ} {c : ℂ} {f : ℂ → F} (hR : 0 < R)
+  (hc : continuous_on f (closed_ball c R)) (hd : differentiable_on ℂ f (ball c R)) :
+  deriv f c = (2 * π * I : ℂ)⁻¹ • ∮ z in C(c, R), (z - c) ^ (-2 : ℤ) • f z :=
+begin
+  lift R to ℝ≥0 using hR.le,
+  refine (has_fpower_series_on_ball_of_continuous_on_of_differentiable_on
+    hc hd hR).has_fpower_series_at.deriv.trans _,
+  simp only [cauchy_power_series_apply, one_div, zpow_neg₀, pow_one, smul_smul,
+    zpow_two, mul_inv₀]
+end
+
+/-- If `f` is continuous on a closed disc of radius `R`, is complex differentiable on its interior,
+and its values on the boundary circle of this disc are bounded from above by `C`, then the norm of
+its derivative at the center is at most `C / R`.
+
+TODO: drop unneeded assumptions `[complete_space F] [second_countable_topology F]`.  -/
+lemma norm_deriv_le_of_forall_mem_sphere_norm_le {c : ℂ} {R C : ℝ} {f : ℂ → F} (hR : 0 < R)
+  (hc : continuous_on f (closed_ball c R)) (hd : differentiable_on ℂ f (ball c R))
+  (hC : ∀ z ∈ sphere c R, ∥f z∥ ≤ C) :
+  ∥deriv f c∥ ≤ C / R :=
+have ∀ z ∈ sphere c R, ∥(z - c) ^ (-2 : ℤ) • f z∥ ≤ C / (R * R),
+  from λ z (hz : abs (z - c) = R), by simpa [norm_smul, hz, zpow_two, ← div_eq_inv_mul]
+    using (div_le_div_right (mul_pos hR hR)).2 (hC z hz),
+calc ∥deriv f c∥ = ∥(2 * π * I : ℂ)⁻¹ • ∮ z in C(c, R), (z - c) ^ (-2 : ℤ) • f z∥ :
+  congr_arg norm (deriv_eq_smul_circle_integral hR hc hd)
+... ≤ R * (C / (R * R)) :
+  circle_integral.norm_two_pi_I_inv_smul_integral_le_of_norm_le_const hR.le this
+... = C / R : by rw [mul_div_comm, div_self_mul_self', div_eq_mul_inv]
+
+/-- **Liouville's theorem**: a complex differentiable bounded function `f : ℂ → E` is a constant.
+Use a more general `complex.apply_eq_apply_of_differentiable_of_bounded` instead. -/
+lemma apply_eq_apply_of_differentiable_of_bounded_dim1 {f : ℂ → F} (hf : differentiable ℂ f)
```

Comment:
For this and the other versions of Liouville's theorem below, why not name them `differentiable.*` so that we have dot notation?

Reply:
Done.